src/renderer/inc/CSSLengthPercentage.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+struct CSSLengthPercentage
+{
+    enum class ReferenceFrame : uint8_t
+    {
+        // This indicates the object is empty/unset.
+        // No need to call Resolve().
+        None,
+        // Call Resolve() with factor set to the target DPI (e.g. 96 for DIP).
+        // Returns an absolute length value scaled by that DPI.
+        Absolute,
+        // Call Resolve() with factor set to the font size
+        // in an arbitrary DPI. Returns a value relative to it.
+        FontSize,
+        // Call Resolve() with factor set to the "0" glyph advance width
+        // in an arbitrary DPI. Returns a value relative to it.
+        AdvanceWidth,
+    };
+
+    static CSSLengthPercentage FromString(const wchar_t* str);
+
+    ReferenceFrame GetReferenceFrame() const noexcept;
+    float Resolve(float factor) const noexcept;
+    float Resolve(float fallback, float dpi, float fontSize, float advanceWidth) const noexcept;
+
+private:
+    float _value = 0;

```

Comment:
it's a float here, but a double elsewhere..!

Reply:
(Let's discuss this in the other float/double comment above. I think it's the same thing.)